### PR TITLE
Added support for AnyROCArrays for GPUArrays.mapreducedim!.

### PR DESCRIPTION
**Pull request by matinraayai**
Hello @maleadt @jpsamaroo,
On my side, this commit seems to add support for ```AnyROCArray```s in ```GPUArray.mapreducedim!```, and it assumes that the passed array is located on the default device. Please let me know when you're done working on the device API for AMDGPU so that I can add that in.
Matin

### CODE REVIEW

**Comment by jpsamaroo** [comment]
@matinraayai would you mind adding a test for this?

**Comment by matinraayai** [comment]
@jpsamaroo aren't these tested in ```GPUArrays.jl```? 
https://github.com/JuliaGPU/GPUArrays.jl/blob/master/test/testsuite/reductions.jl
I wasn't able to find the corresponding ```CUDA.jl``` for this either.

**Comment by maleadt** [comment]
The specific invocation requiring an AnyROCArray couldn't have been tested in GPUArrays or AMDGPU.jl wouldn't have passed tests before. Ideally we add that test in GPUArrays.jl though.

File: src/mapreduce.jl [hunk 1]
```diff
@@ -130,7 +130,7 @@ function GPUArrays.mapreducedim!(f::F, op::OP, R::ROCArray{T},
     # so that we can span the entire reduction dimension using a single item group.
 
     # group size is restricted by local memory
-    agent = R.buf.agent
+    agent = get_default_agent()
```
Comment (jpsamaroo) [inline src/mapreduce.jl:133]:
```suggestion
    agent = device(R)
```

**Review by jpsamaroo — APPROVED** [review]
Now that AMDGPU 0.4 is released, we can support this properly!

**Comment by jpsamaroo** [comment]
@matinraayai you'll need to rebase on AMDGPU master so that you get the `device` call.

**Comment by jpsamaroo** [comment]
@matinraayai bump

**Comment by jpsamaroo** [comment]
Seems like you accidentally force-pushed the wrong branch here @matinraayai

**Comment by matinraayai** [comment]
@jpsamaroo I forgot I had this PR from master. Not sure how to revert it.

**Comment by matinraayai** [comment]
I'm going to close the PR since I have rocSparse things in my fork's master branch. 
Link to new PR: https://github.com/JuliaGPU/AMDGPU.jl/pull/291

**Pull request closed without merging**